### DISCUSSION TITLE
Moving the server to a remote panel

### DISCUSSION THREAD

**Discussion by CodeExplorerX** [pelican-dev/panel#837] [Ideas]
It would be nice to be able to move the server from the control panel to another panel. What would it look like? 

1. The user goes to the server migration section
2. Selects the server that you want to transfer
3. Enters the remote control panel address as well as the user's API key.
4. After successful authorization, the user selects the server to which the files will be transferred. 
5. A backup copy of the server is created at the API level and this copy is transferred to a remote server where it will be unzipped.

**Additions:** 
To the above, we could also add the possibility of migrating a database that would work by a similar analogy. The user selects a local and remote database, creates a copy of the local database and transfers it to the remote database.

**Notes:** 
• In the second paragraph, if the server is turned on, then suggest that the user turn it off to transfer the server without data loss

**Comment by rmartinoscar** [comment]
Sounds good but we need to re implement Server Transfers first.
You need an Application API Key in order to create a server unless it already exists on the remote panel which we'll also need to copy startup, yold, egg variables, etc... which would be a much bigger mess with a user api key then a single create server query as admin.

Migrating the db won't ever be fool proof, if for some reason the php execution time is too little for a heavy db it will fail; same if for what ever reason the collation isn't the supported by the new database host.

**Reply by CodeExplorerX** [reply]
My idea mostly assumed that the user already has a server with a similar environment, and he only needs to transfer files.
If we go beyond this and assume that we also need to create a server from scratch with a similar configuration, then after successful authorization, add the item "send a request to create a server" in paragraph 4 and the remote hosting administrator in the administration panel could approve or reject this request. The information about the request could contain server characteristics (RAM, CPU, Storage), information about the egg and the startup command. 

Regarding the database. Then it would be possible to implement its transfer at the level of the wings. And before sending it, pack it into an archive for faster transfer and without much load. And also in .env or another place for configuration, implement parameters to limit the size of servers and databases to be migrated.

**Reply by CodeExplorerX** [reply]
Regarding database compatibility, based on the Pterodactyl documentation, in most cases it is MariaDB of new versions, and the chance that there will be compatibility problems is unlikely. But it's still real, I don't know if it's possible, but at the software level, somehow determine the version as well as the name of the database server. And add compatibility check when migrating
